src/components/itemContextMenu.js

```diff
@@ -346,6 +355,48 @@ import toast from './toast/toast';
                         getResolveFunction(getResolveFunction(resolve, id), id)();
                     });
                     break;
+                case 'downloadall': {
+                    const downloadEpisodes = episodes => {
+                        import('../scripts/fileDownloader').then((fileDownloader) => {
+                            const downloads = episodes.map(episode => {
+                                const downloadHref = apiClient.getItemDownloadUrl(episode.Id);
+                                return {
+                                    url: downloadHref,
+                                    itemId: episode.Id,
+                                    serverId: serverId,
+                                    title: episode.Name,
+                                    filename: episode.Path.replace(/^.*[\\/]/, '')
+                                };
+                            });
+
+                            fileDownloader.download(downloads);
+                        });
+                    };
+                    const downloadSeasons = seasons => {
+                        Promise.all(seasons.map(seasonItem => {
+                                return apiClient.getEpisodes(seasonItem.SeriesId, {
+                                    seasonId: seasonItem.Id,
+                                    userId: options.user.Id,
+                                    Fields: 'CanDownload,Path'
+                                });
+                            }
+                        )).then(seasonData => {
+                            downloadEpisodes([].concat.apply([], seasonData.map(season => season.Items)));
```

Comment:
It looks like we should be able to use `array.flat()` since we include the core js polyfills.

```suggestion
                            downloadEpisodes(seasonData.map(season => season.Items).flat());
```